### PR TITLE
GLO: Add ITS-TPC/ITS Check for QC

### PR DESCRIPTION
**Pull request by f3sch**
This PR adds a check for the ITS-TPC/ITS matching efficiency. 

Configurable values are:
 - minPt ---> minimum Pt for checked range
 - maxPt ---> maximum Pt for checked range
 - threshold ---> conservative Threshold value

Below are two example plots for very low MC statistics, where the only difference is the set threshold.
[good.pdf](https://github.com/AliceO2Group/QualityControl/files/14981460/good.pdf)
[bad.pdf](https://github.com/AliceO2Group/QualityControl/files/14981468/bad.pdf)
P.S. Also the line extending past the canvas is an artefact of low statistics shown here.

### CODE REVIEW

**Comment by f3sch** [comment]
@knopers8 thanks for the feedback!

**Comment by knopers8** [comment]
> If you have nothing else to add I would ping RC so that they can also give feedback before merging? Cheers.

Yes, of course! @aferrero2707 @rmunzer ?

**Comment by rmunzer** [comment]
> > If you have nothing else to add I would ping RC so that they can also give feedback before merging? Cheers.
> 
> Yes, of course! @aferrero2707 @rmunzer ?

HI @knopers8 @f3sch ,

thank you very much. This checker is indeed very useful.
May i ask if it is possible to use the checker also for eta and phi plots?

Thanks 
Robert

**Comment by f3sch** [comment]
Hi @rmunzer, thanks for the feedback. I would suggest to merge this, while we take a look how to define the check for eta/phi. Cheers Felix

**Review by knopers8 — APPROVED** [review]

File: Modules/GLO/src/ITSTPCmatchingCheck.cxx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+using namespace std;
+using namespace o2::quality_control;
+
+namespace o2::quality_control_modules::glo
+{
+
+void ITSTPCmatchingCheck::configure()
+{
+  if (auto param = mCustomParameters.find("minPt"); param != mCustomParameters.end()) {
```
Comment (aferrero2707) [inline Modules/GLO/src/ITSTPCmatchingCheck.cxx:45]: I would suggest to move the parameters initialization in the `startOfActivity()` function, and use the extended parameters to allow defining beam-type dependent thresholds, if needed.

File: Modules/GLO/src/ITSTPCmatchingCheck.cxx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    (void)moName;
+
+    if (mo->getName() == "mFractionITSTPCmatch_ITS") {
+      auto* eff = dynamic_cast<TEfficiency*>(mo->getObject());
+      auto binLow = eff->FindFixBin(mMinPt), binUp = eff->FindFixBin(mMaxPt);
+
+      result = Quality::Good;
+
+      if (eff->GetTotalHistogram()->GetEntries() == 0) {
+        result = Quality::Bad;
```
Comment (aferrero2707) [inline Modules/GLO/src/ITSTPCmatchingCheck.cxx:88]: Why not set the quality to `Null` if the denominator is empty? In such case the efficiency cannot be determined.

File: Modules/GLO/src/ITSTPCmatchingCheck.cxx [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+        result.addReason(FlagReasonFactory::Unknown(), "No ITS tracks in denumerator!");
```
Comment (aferrero2707) [inline Modules/GLO/src/ITSTPCmatchingCheck.cxx:89]: `denumerator` -> `denominator`

File: Modules/GLO/src/ITSTPCmatchingCheck.cxx [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    } else {
+      msg->SetFillColor(kRed);
+      for (const auto& [_, desc] : checkResult.getReasons()) {
+        msg->AddText(Form("%s", desc.c_str()));
+      }
+      msg->SetTextColor(kWhite);
+    }
+    eff->GetListOfFunctions()->Add(msg);
+
+    if constexpr (false) {
```
Comment (aferrero2707) [inline Modules/GLO/src/ITSTPCmatchingCheck.cxx:219]: This should be probably removed in the final version

**Comment by rmunzer** [comment]
> Hi @aferrero2707, @rmunzer,
> 
> I addressed your suggestion. There are now more options moved to extendedCheckParameters:
> 
> * showPt : default true
> * thresholdPt
> * min/maxPt
> * showPhi: default true     <----- might be useful to spot TPC sector/ITS stave issues
> * thresholdPhi
> * showEta: default false    <----- it is there, but we did not come up with what insight it can provide
> * thresholdEta
> * min/maxEta
> 
> I attach two example bad quality plots. [testPhi.pdf](https://github.com/AliceO2Group/QualityControl/files/15024915/testPhi.pdf) [testEta.pdf](https://github.com/AliceO2Group/QualityControl/files/15024916/testEta.pdf)
> 
> Any feedback is appreciated :)

Very nice. 
Thank you very much for adding those two check. 
In think all three will be very useful.

**Pull request merged**